docs/user/rstudio/ide/guide/tools/copilot.qmd

```diff
@@ -0,0 +1,237 @@
+  }
+  sqrt(var(x) / length(x))
+}
+```
+
+Which generated a working example!
+
+``` r
+calc_se(1:10)
+#> [1] 0.9574
+
+# note the function also properly indicates the need for a numeric input
+calc_se(letters)
+#> Error in calc_se(letters) : x must be numeric
+```
+
+For smaller problems or to provide context or guidance within a function or code block, comments can be used to provide a basic prompt for Copilot.
+
+Comments in code such as:
+
+``` r
+# create a function that takes a number and returns the square
+```
+
+Might be used by Copilot to suggest a function that takes a number and returns the square:
+
+``` r
+# create a function that takes a number and returns the square
+square_number <- function(x) {
+  x^2
+}
+```
+
+Lastly, comments can be used inline within code to provide additional context or guidance to Copilot. For example:
+
+``` r
+library(dplyr)
+
+mtcars |> 
+   group_by(cyl) |> 
+   summarize(
+      mean = mean(mpg),
+      # add the standard error for mpg
+   )
+```
+
+Might return:
+
+``` r
+mtcars |> 
+   group_by(cyl) |> 
+   summarize(
+      mean = mean(mpg),
+      # add the standard error for mpg
+      se = sd(mpg) / sqrt(n())
+   )
+```
+
+However, the comment may not needed if a descriptive column name is used instead.
+
+![](images/copilot_se-inline.png){fig-alt="Copilot inline completion"}
+
+### Language support
+
+While R is the primary expected language for RStudio users, Copilot also supports other languages such as SQL, Python, HTML, CSS, JavaScript, and many other languages. Copilot will use the file extension or other context to determine the language of the active document. In multi-language Quarto documents for example, Copilot could use the active language within a code chunk or the body of the document to generate suggestions.
+
+Comments in other languages can be used to generate suggestions in the active document For example, a HTML comment (`<!-- COMMENT BODY -->`) might be used in the body of a `.qmd` file:
+
+``` html
+<!-- Using HTML and JavaScript return a clickable counter that is displayed -->
+<!-- there should also be a button to reset the counter to zero -->
+```
+
+Where the above HTML comment might generate the below HTML and JavaScript:
+
+``` html
+<!-- Using HTML and JavaScript return a clickable counter that is displayed -->
+<!-- there should also be a button to reset the counter to zero -->
+<html>
+<body>
+<script>
+var count = 0;
+function countClicks() {
+  count += 1;
+  document.getElementById("clicks").innerHTML = count;
+}
+function resetClicks() {
+  count = 0;
+  document.getElementById("clicks").innerHTML = count;
+}
+</script>
+<p>Click the button to count clicks.</p>
+<button type="button" onclick="countClicks()">Click Me!</button>
+<button type="button" onclick="resetClicks()">Reset</button>
+<p>Clicks: <a id="clicks">0</a></p>
+</body>
+</html>
+```
+
+![](images/copilot_javascript.png){fig-alt="Copilot HTML and JavaScript completions"}
+## Support
```

Comment:
Need a blank line above this, before the image, or else this and the below show up as text instead of a blurb.

Reply:
done